### PR TITLE
Closes #2081: Upgrade Drupal core to 10.0.9

### LINKED ISSUE #2081: Drupal 10 compatibility
## What is the problem that we want to solve?
We need to prepare Quickstart for a Drupal 10 upgrade.  This will require verifying the Drupal 10 compatibility of all of the contrib projects we use as well.

## Resources
Acquia's Drupal 10 project deprecation status tracker:
https://dev.acquia.com/drupal10/deprecation_status/projects

## Conditions of satisfaction
- [ ] The considered options are documented
- [ ] A recommended next step is documented
- [ ] An estimate of time and effort is provided for the recommendation
- [ ] …

### PR DESCRIPTION
**Pull request by joeparsons**
## Description
- Upgrades Drupal core to 10.0.9
- Adds CKEditor 4 contrib module
- Upgrades various contrib modules to D10-compatible versions (including `config_sync`/`config_distro` suite of modules)
- Updates core and contrib patches
- Fixes various issues with tests
  - Replaces usage of `seven` theme in tests with `claro` since `seven` is no longer available in D10
  - Adds new patch for [Bootstrap Barrio JS library issue](https://www.drupal.org/project/bootstrap_barrio/issues/3366908) that was causing an error that broke a PHPUnit test
  - Updates usage of `str_replace` migrate process plugin provided by `migrate_plus` in various migrations to fix failing migration tests
- Fixes various issues reported by phpstan and PHPCS
- Fixes an issue that @tadean noticed involving once.js that affected the Event Calendar and Publication date picker
- Updates lando and ProboCI configs
- Removes lando/ddev tooling for Drupal console
- Updates phpstan and PHPCS configs
- Replaces `drupal/background_image_formatter` and `drupal/smart_title` with az-digital D10-compatible forks 

### Related PRs
These will need to be merged in conjunction with this one:
- Scaffolding Repo PR: https://github.com/az-digital/az-quickstart-scaffolding/pull/104
- Dev Dependencies Metapackage PR: https://github.com/az-digital/az-quickstart-dev/pull/82
- Pantheon Upstream PR: https://github.com/az-digital/az-quickstart-pantheon/pull/165 (the timing of this one will need to be very specific and probably in conjuction with the final 2.7.0 stable release)

Some of the changes included here are also available in individual PRs where they may be easier to review/test/merge in isolation (and probably should be):
- https://github.com/az-digital/az_quickstart/pull/2452
- https://github.com/az-digital/az_quickstart/pull/1999
- https://github.com/az-digital/az_quickstart/pull/2325
- https://github.com/az-digital/az_quickstart/pull/2450
- https://github.com/az-digital/az_quickstart/pull/2489

## Related issues

Closes #2081 

## How to test

- General testing
- CKeditor testing

### Arizona Quickstart (install profile, custom modules, custom theme)
- **Patch release changes**
   - [ ] Bug fix
   - [ ] Accessibility, performance, or security improvement
   - [ ] Critical institutional link or brand change
   - [ ] Adding experimental module
   - [ ] Update experimental module
- **Minor release changes**
   - [ ] New feature
   - [ ] Breaking or visual change to existing behavior
   - [ ] Upgrade experimental module to stable
   - [ ] Enable existing module by default or database update
   - [ ] Non-critical brand change
   - [ ] New internal API or API improvement with backwards compatibility
   - [ ] Risky or disruptive cleanup to comply with coding standards
   - [x] High-risk or disruptive change (requires upgrade path, risks regression, etc.)
- **Other or unknown**
   - [ ] Other or unknown

### Drupal core
- **Patch release changes**
   - [ ] Security update
   - [ ] Patch level release (non-security bug-fix release)
   - [ ] Patch removal that's no longer necessary
- **Minor release changes**
   - [x] Major or minor level update
- **Other or unknown**
   - [ ] Other or unknown

### Drupal contrib projects
- **Patch release changes**
   - [ ] Security update
   - [ ] Patch or minor level update
   - [ ] Add new module
   - [ ] Patch removal that's no longer necessary
- **Minor release changes**
   - [x] Major level update
- **Other or unknown**
   - [ ] Other or unknown

## Checklist

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

**Review by trackleft — DISMISSED** [review]

**Comment by tadean** [comment]
It seems like `Storage comparer issue` in the removed `config_distro` patches might still exist (though the patch no longer applies), but I'm not sure...this was for #428 I think . [Here](https://gist.githubusercontent.com/tadean/e2e0dd57f1a9c4eae28d8e713cb01d82/raw/e114987d4a76c183208c763de4dbfecbf41b7f35/config_distro_storage_comparer_issue.patch) is a potentially rerolled patch, though maybe the actual cause could be elsewhere...

**Comment by camikazegreen** [comment]
When attempting to edit [this News article](https://7b5e2985-cedf-4a13-ab91-3a35916c835d--pr-2354.probo.build/news/runners-brains-may-have-more-connectivity-ua-research-shows) as a Content Admin, I get this error:
Screenshot 2023-06-16 at 9 29 49 AM

**Comment by joeparsons** [comment]
> It seems like `Storage comparer issue` in the removed `config_distro` patches might still exist (though the patch no longer applies), but I'm not sure...this was for #428 I think . [Here](https://gist.githubusercontent.com/tadean/e2e0dd57f1a9c4eae28d8e713cb01d82/raw/e114987d4a76c183208c763de4dbfecbf41b7f35/config_distro_storage_comparer_issue.patch) is a potentially rerolled patch, though maybe the actual cause could be elsewhere...

Thanks @tadean I've restored the (re-rolled) patch.

**Review by Carris-Toupin — DISMISSED** [review]

**Review by tadean — APPROVED** [review]

**Comment by tadean** [comment]
After taking a D9 build and moving it to #2354, a cache clear was required to get the site working again. The following database updates were available:
```
 ------------------------ ----------- --------------- ---------------------------------------------------------------- 
  Module                   Update ID   Type            Description                                                     
 ------------------------ ----------- --------------- ---------------------------------------------------------------- 
  better_exposed_filters   8003        hook_update_n   8003 - Provide upgrade path from 8.x-4.x to 8.x-5.x.            
  user                     10000       hook_update_n   10000 - Remove non-existent permissions created by migrations.  
 ------------------------ ----------- --------------- ---------------------------------------------------------------- 
 ```
 
 The following `config_distro` changes were present for import:
 ```
  ---------------- ---------------- ------------------------------------------------ 
  Extension        Operation type   Label                                           
 ---------------- ---------------- ------------------------------------------------ 
  az_core          create           Enterprise Thumbnail                            
  az_core          create           Enterprise Thumbnail Small                      
  az_core          create           taxonomy_term.field_az_attribute_type           
  az_core          create           taxonomy_term.field_az_attribute_key            
  az_core          create           node.field_az_enterprise_attributes             
  az_core          create           Enterprise Attributes                           
  az_core          create           Attribute Type                                  
  az_core          create           Attribute Key                                   
  az_core          create           taxonomy_term.az_enterprise_attributes.default  
  az_core          create           taxonomy_term.az_enterprise_attributes.default  
  az_core          create           Taxonomy List Pages                             
  az_core          update           az_core.settings                                
  az_core          update           smart_title.settings                            
  az_media         update           Standard                                        
  az_news          create           Enterprise Attributes                           
  az_news          create           node.az_news.az_row                             
  az_news          update           node.az_news.default                            
  az_news          update           node.az_news.teaser                             
  az_news          update           node.az_news.default                            
  az_news          update           node.az_news.az_minimal_media_list              
  az_news          update           node.az_news.az_medium_media_list               
  az_news          update           node.az_news.az_media_list                      
  az_news          update           node.az_news.az_marquee                         
  az_news          update           node.az_news.az_feature                         
  az_news          update           node.az_news.az_card                            
  az_person        update           node.az_person.default                          
  az_publication   create           node.az_publication.az_card                     
  az_publication   create           AZ Publications Curate                          
  az_publication   update           node.az_publication.default                     
  az_publication   update           AZ Publications                                 
  node             update           Content                                         
  user             update           Anonymous user                                  
  user             update           Authenticated user                              
  az_quickstart    update           Content editor                                  
  az_quickstart    update           Content administrator                           
 ---------------- ---------------- ------------------------------------------------ 
 ```
(This existing site had `az_publication` already enabled.) 

 The `ckeditor CKEditor 4 (contrib) (Deprecated)` module showed up as enabled, likely because it has the same identifier as the old core equivalent that was in use prior to D10.
 
 CK Editor still seemed to work for me, and had styles and embeds available.
 
ck4

File: composer.json [hunk 1]
```diff
@@ -135,38 +136,36 @@
...
                 "Rendered content view with relationship issue (2457999)": "https://www.drupal.org/files/issues/2023-01-04/2457999-9.5.x-309.patch",
                 "Views Date Filter Datetime Granularity Option (2868014)": "https://www.drupal.org/files/issues/2022-11-16/2868014-127.datetime-granularity.patch",
                 "Unnecessary restrictions on logo format: Can't upload replacement SVG logo (2259567)": "https://www.drupal.org/files/issues/2022-05-05/2259567-160-9-3-x.patch",
-                "Ajax CSS load order issue (1461322)": "https://www.drupal.org/files/issues/2020-06-05/1461322-25.patch",
+                "Ajax CSS load order issue (1461322)": "https://www.drupal.org/files/issues/2023-06-14/1461322-10-0-x-40.patch",
                 "Layout builder revisions issue (3033516)": "https://www.drupal.org/files/issues/2019-06-17/3033516-17.patch",
-                "Allow text field to enforce a specific text format (784672)": "https://www.drupal.org/files/issues/2023-01-05/784672-302.patch",
                 "Unable to select single view display for view settings (2552541)": "https://www.drupal.org/files/issues/2020-11-03/2552541-43.patch",
```
Comment (trackleft) [inline composer.json:162]:
```suggestion
                "Allow text field to enforce a specific text format (784672)": "https://www.drupal.org/files/issues/2023-04-04/784672-drupal-10.0.x-305.patch",
                "Unable to select single view display for view settings (2552541)": "https://www.drupal.org/files/issues/2020-11-03/2552541-43.patch",
```

**Comment by camikazegreen** [comment]
Confirming that the latest version has resolved the issue with editing News.

**Review by trackleft — APPROVED** [review]

**Review by tadean — APPROVED** [review]

**Review by camikazegreen — APPROVED** [review]

**Pull request merged**